pkgs/development/compilers/ghc/head.nix

```
@@ -14,6 +14,7 @@
   # Libdw.c only supports x86_64, i686 and s390x
 , enableDwarf ? stdenv.targetPlatform.isx86 &&
                 !stdenv.targetPlatform.isDarwin &&
+                !stdenv.hostPlatform.isDarwin &&
                 !stdenv.targetPlatform.isWindows
```

What do you think about the following instead:

```suggestion
, enableDwarf ? (stdenv.targetPlatform.isx86 ||
                stdenv.targetPlatform.isS390) &&
                lib.meta.availableOn stdenv.hostPlatform elfutils &&
                lib.meta.availableOn stdenv.targetPlatform elfutils
```

---

Good idea, I like how this is self-documenting. Why the s390 though?

---

as the comment states above dwarf should also be supported on s390x: <https://gitlab.haskell.org/ghc/ghc/-/blob/master/rts/Libdw.c#L339>

---

Thanks, makes sense. Applied :)